ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Reader.hs

```diff
@@ -299,22 +300,25 @@ instructionHelper registry varReader blockComponent fromMaybeSTM CDB{..} = do
     getBlockComponentFromHeader
       :: forall b'. Header blk -> BlockComponent (ChainDB m blk) b' -> m b'
     getBlockComponentFromHeader hdr = \case
-        GetBlock      -> getBlockComponent GetBlock
-        GetRawBlock   -> getBlockComponent GetRawBlock
-        GetHeader     -> return $ return hdr
-        GetRawHeader  -> return $ toLazyByteString $ encodeDisk codecConfig hdr
-        GetHash       -> return $ headerHash hdr
-        GetSlot       -> return $ blockSlot hdr
-        GetIsEBB      -> return $ headerToIsEBB hdr
-        GetBlockSize  -> getBlockComponent GetBlockSize
+        GetBlock        -> getBlockComponent GetBlock
+        GetRawBlock     -> getBlockComponent GetRawBlock
+        GetHeader       -> return $ return hdr
+        GetRawHeader    -> return $ toLazyByteString $ encodeDisk codecConfig hdr
+        GetHash         -> return $ headerHash hdr
+        GetSlot         -> return $ blockSlot hdr
+        GetIsEBB        -> return $ headerToIsEBB hdr
+        GetBlockSize    -> getBlockComponent GetBlockSize
         -- We could look up the header size in the index of the VolatileDB,
         -- but getting the serialisation is cheap because we keep the
         -- serialisation in memory as an annotation, and the following way is
         -- less stateful
-        GetHeaderSize -> return $
+        GetHeaderSize   -> return $
           fromIntegral $ Lazy.length $ toLazyByteString $ encodeDisk codecConfig hdr
-        GetPure a     -> return a
-        GetApply f bc ->
+        GetNestedType n -> return $
+          Short.toShort $ Lazy.toStrict $ Lazy.take (fromIntegral n) $
+          toLazyByteString $ encodeDisk codecConfig hdr
```

Comment:
Wrong, forward to the VolatileDB

Reply:
Add as a comment/TODO: when we replace `GetNestedType` with the proper type, we won't have to forward it anymore.